core/json/src/main/scala/net/liftweb/json/JsonAST.scala

```diff
@@ -73,47 +138,68 @@ object JsonAST {
       case _ => Nil
     }
 
-    /** XPath-like expression to query JSON fields by name. Returns all matching fields.
-      * <p>
-      * Example:<pre>
-      * json \\ "name"
-      * </pre>
-      */
+    /**
+     * Find all children of a `JObject` with the matching name, returning an empty `JObject` is no matches are found.
+     *
+     * Example:
+     *
+     * {{{
+     * json \\ "name"
+     * }}}
+     */
     def \\(nameToFind: String): JValue = {
       def find(json: JValue): List[JField] = json match {
-        case JObject(l) => l.foldLeft(List[JField]()) { 
-          case (a, JField(name, value)) => 
-            if (name == nameToFind) {
-              a ::: List(JField(name, value)) ::: find(value)
-            } else {
-              a ::: find(value)
-            }
-        }
-        case JArray(l) => l.foldLeft(List[JField]())((a, json) => a ::: find(json))
-        case _ => Nil
+        case JObject(fields) =>
+          fields.foldLeft(List[JField]()) {
+            case (matchingFields, JField(name, value)) =>
+              matchingFields :::
+              List(JField(name, value)).filter(_.name == nameToFind) :::
```

Comment:
I have been slowly adopting the indentation of the second line too, specially on code from lift-json when using  `~` mostly because the IDE does it automatically and I noticed people new to scala think it is strange we align them all the left, instead of making the second line have more indentation.